Source/AwsCommonRuntimeKit/http/HTTP2StreamManagerOptions.swift

```diff
@@ -0,0 +1,197 @@
+
+    /// Connection level window control
+    /// Set to true to manually manage the flow-control window of whole HTTP/2 connection.
+    ///
+    /// If false, the connection will maintain its flow-control windows such that
+    /// no back-pressure is applied and data arrives as fast as possible.
+    ///
+    /// If true, the flow-control window of the whole connection will shrink as body data
+    /// is received (headers, padding, and other metadata do not affect the window) for every streams
+    /// created on this connection.
+    /// The initial connection flow-control window is 65,535. It is not controllable.
+    /// Once the connection's flow-control window reaches to 0, all the streams on the connection stop receiving any
+    /// further data.
+    /// The user must call aws_http2_connection_update_window() to increment the connection's // TODO: update
+    /// window and keep data flowing.
+    /// Note: the padding of data frame counts to the flow-control window.
+    /// But, the client will always automatically update the window for padding even for manual window update.
+    public var enableConnectionManualWindowManagement: Bool
+
+    /// (Optional)
+    /// When set, connection will be closed if 5xx response received from server.
+    public var closeConnectionOnServerError: Bool
+
+    /// (Optional)
+    /// The period for all the connections held by stream manager to send a PING in milliseconds.
+    /// If you specify Nil, manager will NOT send any PING.
+    /// Note: if set, it must be large than the time of ping timeout setting.
+    public var connectionPingPeriodMs: Int?
+
+    /// (Optional)
+    /// Network connection will be closed if a ping response is not received
+    /// within this amount of time (milliseconds).
+    public var connectionPingTimeoutMs: Int?
+
+    /// (Optional)
+    /// The ideal number of concurrent streams for a connection. Stream manager will try to create a new connection if
+    /// one connection reaches this number. But, if the max connections reaches, manager will reuse connections to create
+    /// the acquired steams as much as possible.
+    public var idealConcurrentStreamsPerConnection: Int?
+
+    /// (Optional)
+    /// Default is no limit, which will use the limit from the server.
+    /// The real number of concurrent streams per connection will be controlled by the minimal value of the setting from
+    /// other end and the value here.
+    public var maxConcurrentStreamsPerConnection: Int?
+
+    public init(clientBootstrap: ClientBootstrap,
+                hostName: String,
+                port: UInt16,
+                proxyOptions: HTTPProxyOptions? = nil,
+                proxyEnvSettings: HTTPProxyEnvSettings? = nil,
+                socketOptions: SocketOptions = SocketOptions(),
+                tlsOptions: TLSConnectionOptions? = nil,
+                monitoringOptions: HTTPMonitoringOptions? = nil,
+                maxConnections: Int = 2,
```

Comment:
2 connections max 😳
that seems low
maybe just follow the lead of the C API and force the user to pass it in